src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/VsLangProjectPropertiesProvider.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    /// <summary>
+    /// This provider imports <see cref="VSLangProj.VSProject"/> provided by CPS
+    /// and wraps it into an object that implements both <see cref="VSLangProj.VSProject"/> and 
+    /// <see cref="VSLangProj.ProjectProperties"/>. This enables us to provide
+    /// ProjectProperties to the Project Property Pages and maintain Backward Compatibility.
+    /// </summary>
+    internal class VsLangProjectPropertiesProvider
+    {
+        private readonly VSProject _vsProject;
+        private readonly IProjectThreadingService _threadingService;
+        private readonly ActiveConfiguredProject<ProjectProperties> _projectProperties;
+
+        [ImportingConstructor]
+        public VsLangProjectPropertiesProvider(
+            [Import(ExportContractNames.VsTypes.CpsVSProject)] VSProject vsProject,
+            IProjectThreadingService threadingService,
+            ActiveConfiguredProject<ProjectProperties> projectProperties)
+        {
+            Requires.NotNull(vsProject, nameof(vsProject));
+            Requires.NotNull(threadingService, nameof(threadingService));
+            Requires.NotNull(projectProperties, nameof(projectProperties));
+
+            _vsProject = vsProject;
+            _threadingService = threadingService;
+            _projectProperties = projectProperties;
+        }
+
+        [Export(ExportContractNames.VsTypes.VSProject, typeof(VSProject))]
+        [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+        [Order(10)]
```

Comment:
The CPS's implementation is exported with default value(0) . Having an order 10, will give the ability for extensions to export their implementation with an order between 0 &10 and the ability to export with an order greater than 10. We dont want to export our's with `MaxValue` becauseno one else can have their export ordered higher than ours.